File: java-springboot/src/main/java/com/artitk/RESTfulSpringBoot/controller/ItemController.java
package com.artitk.RESTfulSpringBoot.controller;

import com.artitk.RESTfulSpringBoot.bean.*;
import com.artitk.RESTfulSpringBoot.model.Store;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;

@RestController
public class ItemController {
    private Store store;

    public ItemController() {
        store = new Store();
    }

    @GetMapping("/item/")
    public Items getAllItems() {
        return new Items(store.getAllItems());
    }

    @GetMapping("/item/{index}")
    public Item getItem(@PathVariable Integer index) {
        return new Item(store.getItem(index));
    }

    @PostMapping("/item/")
    public void addItem(@RequestBody Item item) {
        store.addItem(item.getItem());
    }

    @PutMapping("/item/{index}")
    public HashMap<String, String> updateItem(@PathVariable Integer index, @RequestBody Item item) {
        String oldItem = store.updateItem(index, item.getItem());

        HashMap<String, String> itemHashMap = new HashMap<>();
        itemHashMap.put("oldItem", oldItem);

        return itemHashMap;
    }

    @DeleteMapping("/item/{index}")
    public HashMap<String, String> removeItem(@PathVariable Integer index) {
        String removeItem = store.removeItem(index);

        HashMap<String, String> itemHashMap = new HashMap<>();
        itemHashMap.put("removeItem", removeItem);

        return itemHashMap;
    }
}

File: kotlin-springboot/src/main/kotlin/com/artitk/RESTfulSpringBoot/controller/ItemController.kt
package com.artitk.RESTfulSpringBoot.controller

import com.artitk.RESTfulSpringBoot.bean.Item
import com.artitk.RESTfulSpringBoot.bean.Items
import com.artitk.RESTfulSpringBoot.exception.NotFoundException
import com.artitk.RESTfulSpringBoot.model.Store
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class ItemController {
    private var store: Store? = null

    init {
        store = Store()
    }

    @GetMapping("/item/")
    fun getAllItems(): Items {
        return Items(store!!.getAllItems())
    }

    @GetMapping("/item/{index}")
    fun getItem(@PathVariable index: Int): Item {
        val item: String
        try {
            item = store!!.getItem(index)
        } catch (e: IndexOutOfBoundsException) {
            throw NotFoundException(e)
        }

        return Item(item)
    }
}
